helper_scripts/config_history.py

```diff
@@ -0,0 +1,38 @@
+import subprocess
+from pathlib import Path
+
+def get_last_commit_id(file_path):
+    result = subprocess.run(
+        ["git", "log", "-1", "--format=%H", "--", str(file_path)],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        check=True
+    )
+    return result.stdout.strip()
+
+def get_version_from_commit(commit_id):
+    result = subprocess.run(
+        ["git", "describe", "--tags", commit_id],
```

Comment:
Just so ya know, this is what the dynamic versioning does

Reply:
Yeah, I just stuck together git commands because I couldn't instantly figure out how to get the dynamic version for the last time the file was updated, and I wasn't thinking about merging this anyway.